core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -372,6 +372,8 @@ object SparkSubmit {
       OptionAssigner(args.files, YARN, CLUSTER, clOption = "--files"),
       OptionAssigner(args.archives, YARN, CLUSTER, clOption = "--archives"),
       OptionAssigner(args.jars, YARN, CLUSTER, clOption = "--addJars"),
+      OptionAssigner(args.principal, YARN, CLUSTER, clOption = "--principal"),
```

Comment:
There isn't one - I just didn't know if it would be useful. I don't have any issues with making it available in client mode too.